### ISSUE TITLE
[Question] Append prefix to filename

### ISSUE THREAD

**Issue by akdasa** [pytest-dev/pytest-cov#305]
Good getlocaltime()!

Is it possible to append some prefix to a generated report?
```xml
/// cov.xml
<class filename="{PREFIX_HERE}main.py" line-rate="1" name="main.py">
```

I'm running  coverage like this:
```bash
pytest --cov-report=xml --cov=.
```

Why I need it? I'm using [codecov](https://codecov.io/) and multiple projects in one repo, so I want to distinguish these projects for codecov.

Here is an example:
```
monorepo:
  project1:
    main.py
  project2:
    main.js
```

So I have two coverage reports:
```
/// for codecov there are two files in a root folder. But it's actually not. 
<class filename="{WANT_PREFIX_HERE}main.js" line-rate="1" name="main.js">
<class filename="{WANT_PREFIX_HERE}main.py" line-rate="1" name="main.py">
```

In order to distinguish them, I want to add some prefixes.
Is it possible to do it somehow? Or is there a better way? (Sorry for my terrible English :) )

**Comment by blueyed** [comment]
Do you run pytest only once for the monorepo?
Otherwise you could run it per project, and upload it separately to codecov (using flags).
This would also allow you to split it into 2 CI jobs to be run in parallel.

**Comment by akdasa** [comment]
I run coverage tests twice:
1. pytest for backend (python)
2. jest for frontend (js)
That's why I'm unable to run coverage from the root of a project.

pytest generates relative path:
```
<class filename="main.py" line-rate="1" name="main.py">
```

but, jest generates an absolute path:
```
<file name="LoginView.vue" path="/workspaces/gem-client/src/modules/auth/views/LoginView.vue">
```

So my final report looks like:

<img width="1160" alt="gem-app - Codecov 2019-07-05 10-00-07" src="https://user-images.githubusercontent.com/22368151/60704013-f11a4c80-9f0b-11e9-9ad8-f191d63aacf2.png">

> Otherwise you could run it per project, and upload it separately to codecov (using flags).
I'll try it again, but yestergay I have no luck with it.

**Comment by blueyed** [comment]
Ah, misunderstood it then - I thought project1 and project2 where run together once with pytest.

A simple solution might be to just use `sed` or similar to adjust the generated coverage.xml - either add the prefix, or remove it from the other report.

pytest-cov is unlikely to help here, since it only wraps coverage.py, and AFAIR there is no method there to prepend a path prefix.

**Comment by blueyed** [comment]
Since this is with codecov, I think they also have a mechnism to adjust paths (via codecov.yml or their UI).

**Comment by akdasa** [comment]
I already thought about ```sed``` or something like that. But it looks like little bit hacky.

Codecov has mechanism to fix path:
```
# .codecov.yml
fixes:
  - "workspaces/gem-client/::gem/clients/gem/"
  - "app/::gem/apps/gem/" # will not work, paths are relative and not prefixed with app/
  - "::/gem/apps/gem" # will affect all files in all projects
```
but because paths is relative, I'm unable to fix it and do not affect files that belong to another project.

Thank you for your help. I'll return back with some solution (even with ```sed``` (for googlers) :) )

**Comment by akdasa** [comment]
Here a "solution":
1. Generate ```.coverage``` report: ```pytest pytest --cov=.```. It will produce report with absolute paths.
2. Move ```.coverage``` to a parent folder.
3. Generate report ```coverage xml```

Here is a how I fixed it for myself.
```
$ pwd
/app
$ pytest --cov=. && cd .. && cp ./app/.coverage . && coverage xml -o /coverage/coverage.xml
```

**Comment by ionelmc** [comment]
@akdasa so this is solved/can be closed?

**Comment by akdasa** [comment]
@ionelmc Yes, please.
